Changelogs for Digital Twin Manager UI 1023.0.0 (#3879)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/dtm-1023-0-0-added-permission-check-when-creating-updating-deleting-an-asset-based-on-c-8-y-external-asset-assets-permission-mode.md

---
date: ""

> title: "Added permission check when creating, updating, or deleting an asset"

Reviewer: ```suggestion
title: "Enabled fine-grained permissions for managing assets"
```

> title: "Added permission check when Creating/Updating/Deleting an Asset based on c8y_ExternalAsset & assets.permission.mode"

Reviewer 2: ```suggestion
title: "Added permission check when creating, updating, or deleting an asset"
```

product_area: "Application enablement & solutions"
change_type:
    - value: "change-QHu1GdukP"
      label: "Feature"
component:
    - value: "component-Tl88RYb4A"
      label: "Digital Twin Manager"
build_artifact:
    - value: "tc-wYIY0MBDO"
      label: "dtm"

> ticket: ""

Reviewer: ```suggestion
ticket: "CTM-2525"
```

version: "1023.0.0"
---

> The digital twin manager application user will need additional

Reviewer 2: ```suggestion
The Digital Twin Manager application now requires additional
```

> permissions based on a new tenant option `assets.permission.mode`. The new tenant option

Reviewer: ```suggestion
permissions based on a new tenant option `assets.permission.mode` to manage assets. The new tenant option
```

> permissions based on a new tenant option "assets.permission.mode".It

Reviewer 2: ```suggestion
permissions based on a new tenant option `assets.permission.mode`. The new tenant option
```

> supports 3 different modes: external (default), all and none. The user

Reviewer 2: What do the modes mean? In the title you mention create, update, delete but in the text this is not mentioned. Can we align this, explaining which permission covers which actions, especially "external"?

Reviewer: modes are for permissions. The tenant option can take values: external, all or none. Does the suggestion in my next comment suffice?

Reviewer 2: I understand this but what does it mean if I have the permission "external"? What am I allowed to do then? As I said, in the title you mention various actions but they do not show up here in the text.

Reviewer: Based on your suggestion, I've modified the changelog content. Please review

Reviewer 2: This is much clearer, thanks.

always requires the corresponding ROLE_INVENTORY_* permission

> irrespective of the mode and the DIGITAL_TWIN_ASSETS permission set.

Reviewer: ```suggestion
The Digital Twin Manager now supports fine-grained permission control for managing assets by introducing a new tenant option `assets.permission.mode`.  
This tenant option supports three modes:
- external (Default): Digital twin assets permissions are only enforced for assets marked with the **c8y_ExternalAsset** key. Users require the Digital twin assets Update or Admin permission to modify or delete these assets.
- all: Digital twin assets permissions are strictly required for all assets. Users must possess the Digital twin assets Create permission for creation, the Update permission for modification, and the Admin permission for full management (create, update, or delete) of any asset.
-none: Digital twin assets permissions are not applicable to any assets. Asset control defaults to standard Inventory permissions.
```

> - external (Default): Digital twin assets permissions are only enforced for assets marked with the **c8y_ExternalAsset** key. Users require the Digital twin assets Update or Admin permission to modify or delete these assets.

Reviewer 2: ```suggestion
- external (Default): Digital twin assets permissions are only enforced for assets marked with the **c8y_ExternalAsset** key. Users require the Digital twin assets UPDATE or ADMIN permission to modify or delete these assets.
```

> - all: Digital twin assets permissions are strictly required for all assets. Users must possess the Digital twin assets Create permission for creation, the Update permission for modification, and the Admin permission for full management (create, update, or delete) of any asset.

Reviewer 2: ```suggestion
- all: Digital twin assets permissions are strictly required for all assets. Users must possess the Digital twin assets CREATE permission for creation, the UPDATE permission for modification, and the ADMIN permission for full management (create, update, or delete) of any asset.
```

> -none: Digital twin assets permissions are not applicable to any assets. Asset control defaults to standard Inventory permissions.

Reviewer 3: ```suggestion
- none: Digital twin assets permissions are not applicable to any assets. Asset control defaults to standard Inventory permissions.
```
